### DISCUSSION TITLE
[@types/google-apps-script] Missing types for event objects passed to add-on triggers

### DISCUSSION THREAD

**Discussion by willdhorn** [DefinitelyTyped/DefinitelyTyped#69664] [Issues with a @types package] [answered]
There are no types for the objects that are passed to the top-level functions used as triggers and actions by add-ons.

The type definitions aren't defined under "Reference" in the docs, so struggled a bit trying to find them, but they do show up in one of the guides for using cards: https://developers.google.com/apps-script/add-ons/concepts/event-objects#gmail-event-object.

[a comment by an automated account is not shown]

**Comment by mtgto** [comment, marked as answer]
@willdhorn Is this what you are looking for: `GoogleAppsScript.Addons.GmailEventObject`?

```ts
const handleGmailEventAction = (e: GoogleAppsScript.Addons.GmailEventObject) => {
    Logger.log(e.accessToken);
}
```

**Comment by willdhorn** [comment]
Yes, thank you. Sorry, I could have sworn I looked at everything; I even grep'd for 'accessToken' and it came up empty, but clearly the problem was between my keyboard and chair.
